head.go

```diff
@@ -150,6 +152,14 @@ func newHeadMetrics(h *Head, r prometheus.Registerer) *headMetrics {
 		Name: "prometheus_tsdb_head_samples_appended_total",
 		Help: "Total number of appended samples.",
 	})
+	m.checkpointDeleteFail = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "prometheus_tsdb_checkpoint_deletions_failed_total",
+		Help: "Total number of checkpoint deletions that failed.",
+	})
+	m.checkpointsFail = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "prometheus_tsdb_checkpoints_failed_total",
+		Help: "Total number of checkpoints that failed.",
+	})
```

Comment:
I don't think we have a use case that required this metric and I prefer to add lines of code only when we need to.

Reply:
The initial [request](https://github.com/prometheus/prometheus/issues/4578) from @beorn7 was

> I could not find any metric that got incremented for a failure to truncate the WAL. That would at least allow to alert on a non-truncating WAL before the disk runs full.

This is what the new `prometheus_tsdb_wal_truncate_failed_total` metric is about but it doesn't cover for errors in `head.Truncate()`. So I would say that we need only one additional metric `prometheus_tsdb_head_truncate_total` that is incremented every time `head.Truncate()` fails. The exact cause (checkpoint deletion failed or any other file system error) can be determined by looking at the logs.

Reply:
If I understand correctly, we are looking at only 2 metrics in whole, `prometheus_tsdb_wal_truncate_failed_total` and `prometheus_tsdb_head_truncate_failed_total`, correct?

But another thing is that, failing of `DeleteCheckpoints(...)` is not considered as head truncate fail, so do we ignore it?

Reply:
> But another thing is that, failing of DeleteCheckpoints(...) is not considered as head truncate fail, so do we ignore it?

Right, I missed that part. Then we can keep `prometheus_tsdb_checkpoint_deletions_failed_total`.

Reply:
@krasi-georgiev I have removed `prometheus_tsdb_checkpoints_failed_total`, rather added `prometheus_tsdb_head_truncate_failed_total`. Now these 3 metrics should cover `head.Truncate(...)` entirely.

Reply:
@simonpasquier The main idea is to know when we have a failure that results in increasing the used hdd space which is when a checkpoint deletion fails or when the wal truncation fails.
In this regard why do you think we need to know about `prometheus_tsdb_head_truncate_total`? 
What could we use this metric for?